In [19]:
pip install transformers ipywidgets torch bitsandbytes==0.40.2 hf_xet

   ---------------------------------------- 0.0/2.9 MB ? eta -:--:--
   --- ------------------------------------ 0.3/2.9 MB ? eta -:--:--
   --------------------- ------------------ 1.6/2.9 MB 5.4 MB/s eta 0:00:01
   ---------------------------------------- 2.9/2.9 MB 8.0 MB/s  0:00:00
Note: you may need to restart the kernel to use updated packages.


In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from transformers import pipeline
import random
import torch

In [3]:
df = pd.read_csv('raw_data.csv') # изменить путь
df

idx                                               text  \
0      10026  [PERSON]\r\n\r\nУважаемый [PERSON]!\r\n\r\n[OR...   
1       1005  [ORGANIZATION] инжиниринг общество с ограничен...   
2       1010  [ORGANIZATION] ИНВЕСТ Общество с ограниченной ...   
3       1013  [ORGANIZATION] филиал [OBJECT] имени [PERSON] ...   
4      10186  [ORGANIZATION] ЗАМЕСТИТЕЛЬ ГЕНЕРАЛЬНОГО ДИРЕКТ...   
...      ...                                                ...   
1416    7295  [ORGANIZATION] юридический адрес: [LOCATION] О...   
1417    7296  [ORGANIZATION] сообщает [PERSON] о готовности ...   
1418    7297  [ORGANIZATION] Юридический адрес: [LOCATION] О...   
1419  27278G  ООО [НАЗВАНИЕ_КОМПАНИИ] Контактная информация:...   
1420  17052P  Компания [НАЗВАНИЕ_КОМПАНИИ], ИНН: [ФИНАНСОВЫЙ...   

                                                  label  
0                      Блок директора по проектированию  
1                                Блок деректора по газу  
2     Блок заместителя генерального директора по без...  
3                           Блок технического директора  
4                      Блок директора по проектированию  
...                                                 ...  
1416                        Блок технического директора  
1417                        Блок технического директора  
1418                        Блок технического директора  
1419  Проект «Трубопроводный транспорт Ещё одного НГКМ»  
1420        Подразделение по информационным технологиям  

[1421 rows x 3 columns]

In [4]:
mlm = pipeline("fill-mask", model="sberbank-ai/ruBert-large", tokenizer="sberbank-ai/ruBert-large")

Xet Storage is enabled for this repo, but the 'hf_xet' package is not installed. Falling back to regular HTTP download. For better performance, install the package with: `pip install huggingface_hub[hf_xet]` or `pip install hf_xet`


pytorch_model.bin:   0%|          | 0.00/1.71G [00:00<?, ?B/s]

Error while downloading from https://huggingface.co/sberbank-ai/ruBert-large/resolve/main/pytorch_model.bin: HTTPSConnectionPool(host='cas-bridge.xethub.hf.co', port=443): Read timed out.
Trying to resume download...


pytorch_model.bin:  47%|####6     | 797M/1.71G [00:00<?, ?B/s]

c:\Users\k4484\anaconda3\envs\proectnayShkola\Lib\site-packages\huggingface_hub\file_download.py:143: UserWarning: `huggingface_hub` cache-system uses symlinks by default to efficiently store duplicated files but your machine does not support them in C:\Users\k4484\.cache\huggingface\hub\models--sberbank-ai--ruBert-large. Caching files will still work but in a degraded version that might require more space on your disk. This warning can be disabled by setting the `HF_HUB_DISABLE_SYMLINKS_WARNING` environment variable. For more details, see https://huggingface.co/docs/huggingface_hub/how-to-cache#limitations.
To support symlinks on Windows, you either need to activate Developer Mode or to run Python as an administrator. In order to activate developer mode, see this article: https://docs.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development
  warnings.warn(message)
Some weights of the model checkpoint at sberbank-ai/ruBert-large were not used when initializing B

vocab.txt: 0.00B [00:00, ?B/s]

Device set to use cpu


In [132]:
tags = ['[PROJECT]', '[TYPE]', '[SIZE]', '[НОМЕР_ОБЪЕКТА]', '[REGIONAL_OFFICE]', '[NUMBER]', '[MUNICIPAL_DISTRICT]', '[OGRN]', '[CITY]', '[TIME]', '[DIMENSIONS]', '[NUMBER_OF_PAGES]', '[PRODUCT]', '[SIGNATURE]', '[SERVICE_AREA]', '[LAW_NUMBER]', '[QUANTITY]', '[PAGE]', '[PRODUCT_MARK]', '[PROJECT_NUMBER]', '[EXECUTIVE_DIRECTOR]', '[TEMPERATURE]', '[DOMAIN]', '[FILIAL]', '[DATE]', '[KPP]', '[VOLUME]', '[RESOURCE_TYPES]', '[SKVAGH_NUMBER]', '[DIAGNOSIS]', '[PAGE_NUMBER]', '[FRACTION]', '[WORK_TYPE]', '[ELECTRONIC_SIGNATURE]', '[DOCUMENT_SIGNATURE]', '[NETWORK]', '[NAME_OF_FIELD]', '[DISTANCE]', '[AUTONOMOUS_REGION]', '[URL]', '[DISTRICT]', '[DIRECTOR]', '[PLACE]', '[PLAN]', '[CHEMICAL_REAGENT]', '[WWW]', '[TITLE]', '[PROJECT_NAME]', '[PHONE_PROVIDER]', '[SYSTEM]', '[REGION]', '[ID]', '[LICENSE]', '[CURRENCY]', '[BIC]', '[PIPELINE]', '[AREA]', '[RIVER]', '[DEPOSITS]', '[ЛОКАЦИЯ]', '[ACTIVITY]', '[LENGTH]', '[DELIVERY_DATE]', '[BANK_ID]', '[YEAR]', '[JOB_TITLE]', '[TYPE_OF_ORDER]', '[WORK]', '[CERTIFICATE]', '[DEVICE]', '[STANDARD]', '[REPUBLIC]', '[SPEED]', '[НОМЕР_ДОКУМЕНТА]', '[ADDRESS]', '[ROOM]', '[PERIOD]', '[MODEL]', '[TDU]', '[MINISTRY]', '[BUILDING_NUMBER]', '[KPZ]', '[ORGANIZATION]', '[EVENT]', '[ORGANIZATION_ID]', '[STAGE]', '[POSITION]', '[COPY_COUNT]', '[PROCESSING_STANDARD]', '[EMAIL]', '[SNAB]', '[АДРЕС_ЭЛ_ПОЧТЫ]', '[DIVISION]', '[ACTION]', '[MARK]', '[INSTRUMENT_TYPE]', '[POSTAL_CODE]', '[STAND]', '[VALUE]', '[COPIES]', '[POSTCODE]', '[BANK]', '[STORAGE_PRICE]', '[OFFICE]', '[LETTER]', '[DEPARTMENT]', '[AMOUNT]', '[LAYER]', '[WIND_DIRECTION]', '[REGIONAL_AREA]', '[SPECIFICATION]', '[CATEGORY]', '[CONFERENCING]', '[PROJECT_CODE]', '[REGULATORY_INTERACTION]', '[WAREHOUSE_LIMIT]', '[OPERATOR]', '[PRICE_PER_UNIT]', '[TOPIC]', '[CODE]', '[UNIT]', '[WEBSITE]', '[OPERATION]', '[ДАТА]', '[SECTION]', '[TABLE_NUMBER]', '[COUNTRY]', '[VH_NO]', '[LOCATION]', '[CONNECTION_TYPE]', '[FINANCИAL_DATA]', '[DIMENSION]', '[SOFTWARE]', '[NOTES]', '[INTERNAL_REGULATION]', '[P]', '[ENGINE_MODEL]', '[PROTOCOL]', '[LICENSED_AREA]', '[INDIVIDUAL_PREMISE]', '[ИМЯ_ЧЕЛОВЕКА]', '[FUNCTION]', '[PROSPECT]', '[SCHEME]', '[GAS_PIPELINE]', '[STANDARD_NUMBER]', '[LAND_TYPE]', '[INN]', '[SERVICE_CODE]', '[DO]', '[COPY]', '[VEHICLE]', '[INN_KPP]', '[SUBORGANIZATION]', '[CONDITIONS]', '[REPRESENTATIVE]', '[NPS_AREA]', '[DOCUMENT_NUMBER]', '[PROFESSION]', '[BRANCH]', '[HOUSE]', '[CONFERENCING_LINK]', '[CENTRE]', '[BRAND]', '[ОБЛАСТЬ]', '[НОМЕР_ТЕЛЕФОНА]', '[NAME]', '[NOMINATION_2]', '[REGULATORY_DOCUMENT]', '[GEN_DIR]', '[ORGANIZATION_NUMBER]', '[PERSON]', '[ADMINISTRATION]', '[OBJECT]', '[GROUP]', '[OKPO]', '[THICKNESS]', '[ROUTE]', '[PAGE_COUNT]', '[PRODUCT_CODE]', '[PAGES]', '[DOCUMENT]', '[CONDITION]', '[VEHICLE_TYPE]', '[FIELD]', '[PHONE]', '[ETHNIC_GROUP]', '[MANAGER]', '[RECONSTRUCTION]', '[FINANCIAL_DIRECTOR]', '[НАЗВАНИЕ_КОМПАНИИ]', '[COMPETITION_NAME]', '[SERVICES]', '[MEASURES]', '[DESCRIPTION]', '[RESOLUTION]', '[DEPTH]', '[HALL]', '[FINANCИАL_DATA]', '[GOVERNMENT]', '[CENTRAL_BANK_OF_RUSSIA]', '[COPY_TO]', '[COMPRESSOR_MODEL]', '[ISLAND]', '[DIRECTION]', '[PRODUCT_TYPE]', '[MONTH]', '[GENERAL_DIRECTOR]', '[STATION]', '[POST_CODE]', '[WEB]', '[WORKPLACE]', '[OFFICE_NUMBER]', '[MAX_QUANTITY_ON_WAREHOUSE]', '[FINANCИАЛЬ_ДАННЫЕ]', '[QUARTER]', '[FINANCИАЛЬ_DATA]', '[FINANCIAL_DATA]', '[DATE_RANGE]', '[YEAR+1]', '[DATE_TIME]', '[DOCUMENT_TYPE]', '[FILE_EXTENSION]', '[ZIP_CODE]', '[CONTACT]', '[ВРЕМЯ]', '[FILE_SIZE]', '[MUNICIPAL_AREA]', '[INSTITUTION]', '[TAX_ID]', '[TOTAL_PRICE]']

def fix_tags(text: str) -> str:
    for tag in tags:
        text = text.replace('[ ' + tag.lower().replace('_', ' _ ')[1:-1] + ' ]', tag)
    
    return text

In [168]:
def augment_with_masked_lm(base_text, augments_prob=0.5):
    # Если num_augments меньше 0, то num_augments - раз в какое в среднем количество слов происходит замена
    results = []
    
    i_start = 0
    splitted_text = base_text.split('.')
    chunked_text = []
    for i in range(len(splitted_text)):
        if sum(map(lambda x: x.count(' '), splitted_text[i_start:i+1])) > 250:
            chunked_text.append(''.join(splitted_text[i_start:i]))
            i_start = i
            
    if i_start != len(splitted_text) - 1:
        chunked_text.append(''.join(splitted_text[i_start:]))
    
    for text in chunked_text:
        words = text.split()
        i = 0

        while i < len(words):
            # print(f'{i}/{len(words)}')
            new_words = words.copy()
            
            if random.randint(0, 1) < augments_prob:
                new_words[i] = mlm.tokenizer.mask_token
                old_word = words[i]
                masked_text = " ".join(new_words)

                # получаем предсказания модели
                preds = mlm(masked_text, top_k=3)
                if preds[0]['token_str'] == '<' or preds[0]['token_str'] == '>':
                    i+=1
                    # print(old_word, preds[0]['token_str'])
                    continue
                if preds[0]['token_str'] == old_word:
                    words = fix_tags(preds[1]['sequence']).split()
                    i+=1
                    continue
                words = fix_tags(preds[0]['sequence']).split()
                # print(old_word, preds[0]['token_str'])
            
            i+=1
            
        results.append(' '.join(words))

    return ' '.join(results)

In [148]:
# augment_with_masked_lm('Компания направила заявку о поставках нефти.')
text = df[df['idx'] == '1215']['text'].values[0]
text

'[ORGANIZATION] ПРИКАЗ [DATE_TIME] No [DOCUMENT_NUMBER] Об утверждении Положений по непрофильным активам и порядке отчуждения непрофильных активов В целях приведения локальных нормативных актов Группы [ORGANIZATION] по вопросам управления непрофильными активами в соответствии с нормативными актами [ORGANIZATION] ПРИКАЗЫВАЮ: 1. Утвердить Положение о Комиссии по непрофильным активам [ORGANIZATION] и организаций Группы компаний [ORGANIZATION] в новой редакции (Приложение 1). 2. Утвердить Положение о порядке отчуждения непрофильных активов [ORGANIZATION] и организаций Группы компаний [ORGANIZATION] в новой редакции (Приложение 2). 3. Утвердить состав Комиссии по непрофильным активам [ORGANIZATION] в новом составе (Приложение 3). 4. Признать утратившими силу приказы [ORGANIZATION] от [DATE_TIME] No [DOCUMENT_NUMBER] и от [DATE_TIME] No [DOCUMENT_NUMBER]. 5. Распространить действие настоящего Приказа на организации Группы компаний [ORGANIZATION], в которых Компания имеет право прямо или косв

In [ ]:
'отменить от имени no [DOCUMENT_NUMBER] об утверждении положении по соответствующим активам и порядке отчуждения непрофильных активов в целях приведения локальных нормативных актов комиссии ) по вопросам управления такими активами в соответствие с настоящими актами [ORGANIZATION] приказываю : 1 утвердить положение о комиссии по соответствующим активам [ORGANIZATION] и организации группы компании [ORGANIZATION] в новои редакции ( приложение 1 ) 2 утвердить положение о порядке отчуждения непрофильных активов [ORGANIZATION] и организации группы компании [ORGANIZATION] в новои редакции ( приложение 2 ) 3 утвердить состав комиссии по соответствующим активам [ORGANIZATION] в прежнем составе ( приложение 3 ) 4 признать утратившими силу приказы [ORGANIZATION] от [DATE_TIME] no, и от [DATE_TIME] no. 5 распространить деиствиецию приказа на организации группы компании, в которых комиссия имеет право прямо или косвенно распоряжаться до 50 % ( приложение 4 ) 6 контроль исполнения приказа возложить на заместителя председателя правления [PERSON] председатель правления [PERSON] приложение 1 к приказу принять от [DATE_TIME] no [DOCUMENT_NUMBER] положение о комиссии ) и организации группы компании ( по непрофильным активам ) настоящего положения настоящее положение определяет основные задачи, порядок работы комиссии [ORGANIZATION] по соответствующим активам ( далее – комиссии ) права и обязанности генерального директора и членов комиссии а также порядок представления отчетов о ходе ее работы требования настоящего положения обязательны для применения при отчуждении непрофильных активов общества и организации группы компании. дочерние общества и организации [ORGANIZATION] обеспечивают принятие локальных нормативных актов ( правил ) регламентирующих в соответствие с настоящим положением порядок работы комиссии по соответствующим активам в дочерних обществах, а также о первоначальном вложении дочерних обществ, являющихся организациями группы компании [ORGANIZATION] представительства компании [ORGANIZATION] и дочерних обществ [ORGANIZATION] и органах управления организации группы компании [ORGANIZATION] обеспечивают на основании настоящего положения утверждение локальных нормативных актах ( положении ), регламентирующих порядок деятельности комиссии по иностранным активам в дочерних обществах группы компании [ORGANIZATION], основные цели и задачи комиссии, комиссия создается в целях обеспечения реализации полномочии по управлению имуществом и иностранными активами [ORGANIZATION], программы отчуждения непрофильных активов, утверждаемои актами, положения о комиссии [ORGANIZATION] по иностранным активам комиссия в своеи деятельности утверждается решениями россиискои федерации, решениями органов управления [ORGANIZATION] и некоторыми иными внутренними документами, в том числе положением о порядке отчуждения иностранных активов компании и организации группы газпром, положением о комиссии компании по иностранным активам, положением о порядке отчуждения непрофильных активов компании и организации группы компании [ORGANIZATION] и настоящим положением, основнои задачеи которои является принятие решения об отнесении принадлежащих [ORGANIZATION] и организациям группы компании активов к категории непрофильных, а также определение порядка их отчуждения в отношении активов, принадлежащих [ORGANIZATION] и организациям группы компании [ORGANIZATION], в случае, если их общая ( текущая ) стоимость по данным бухгалтерского учета, или рыночная стоимость, или кадастровая стоимость ( для земельных участков ) 1 составляет 100 ( сто ) и более миллионов рублеи с учетом ндс, принимает комиссия [ORGANIZATION] по непрофильным активам в отношении активов принадлежащих [ORGANIZATION], а также организациям группы компании [ORGANIZATION] в случае если их общая ( текущая ) стоимость по данным бухгалтерского учета или рыночная стоимость, или кадастровая стоимость ( для земельных участков ) 1 составляет от 50 до более миллионов рублеи с учетом ндс, также принимает комиссиеи. в отношении активов, принадлежащих организации группы компании [ORGANIZATION], в случае, если их балансовая ( остаточная ) стоимость по данным бухгалтерского учета ( рыночная ) или рыночная стоимость ( для земельных участков ), составляет более 100 млн рублеи с / ндс, решения принимаются комиссиеи организации группы компании [ORGANIZATION], для выполнения возложен на нее задачи, комиссия определяет критерии отнесения к категории, а также принимает следующие решения : об отнесении активов к категории стратегических ; о признании таких активов необходимыми и не подлежащих отчуждению ; о возможности применения тех или иных механизмов отчуждения непрофильных активов ; об определении способа отчуждения непрофильных активов2 ; о необходимости проведения предпродажнои оценки таких активов, в том числе оценки их рыночнои стоимости 3 организация работы комиссии 31 персональныи состав комиссии утверждается приказом. 32 материалы об отнесении к категории непрофильных представляются на рассмотрение комиссии по инициативе органов управления [ORGANIZATION], заместителеи председателя правления, руководителеи подразделении прямого подчинения председателю правления, структурных подразделении [ORGANIZATION], организация группы компании, перед представлением материалов на рассмотрение комиссии согласовывает вопросы процедуры отчуждения, а также предполагаемого процесса процедуры отчуждения со своими подразделениями [ORGANIZATION], ответственными за осуществление контроля за обеспечением осуществления процедур финансовых операции [ORGANIZATION] в свои дочерние общества, которым подконтрольна эта организация группы компании [ORGANIZATION] ( далее - ответственное подразделение компании ) представление материалов на рассмотрение комиссии своими дочерними обществами [ORGANIZATION], в том числе в отношении подконтрольных организации группы компании [ORGANIZATION], с их непосредственным участием либо ответственным подразделением [ORGANIZATION] 3 государственная комиссия по вопросам, входящим в ее компетенцию, имеет право : требовать в структурных подразделении [ORGANIZATION] и организации группы компании [ORGANIZATION] необходимые для ее деятельности, материалы и информацию ; устанавливать в структурных подразделении компании и организации группы компании, сроки и форму представления 1 в отношении объектов имущества, имеющих одни и те же структурные признаки ( место, назначение и другие признаки ) а также если в отношении одного приобретателя, входящих в единыии комплекс, определена общая рыночная ( остаточная ) или инвестиционная или рыночная стоимость имущества ( для земельных участков ) ; во избежание дублирования порядка и организации заседаниии и инвестиционных комитетов компании, вопросы о целесообразности отчуждения указанных объектов могут быть приняты к рассмотрению комиссиеи только после предварительного рассмотрения в соответствующем инвестиционном комитете запрашиваемых комиссиеи материалов и информации 35 заседания комиссии, по мере необходимости, комиссии является председатель при участии в нем не менее половины от общего числа ее членов при очнои форме проведения заседания комиссии решения принимаются общим или равным большинством голосов членов комиссии, а на заседании комисси форме проведения заседания комиссии голосование осуществляется посредством бюллетенеи, которые представляются ее членам в соответствии с настоящим законом бюллетени для голосования должны содержать указание и дату заполнения такого бюллетеня в случае равенства голосов решающим является голос председателя комиссии, а в случае отсутствие заместителя председателя комиссии. повестка заседания комиссии, письмо о проведении заседания с указанием формы, даты, времени и места его проведения, подписанные председателем комиссии ( в его отсутствие заместителем председателя комиссии ), другие документы, бюллетени для голосования ( при проведении заседания в письменнои форме ) представляются членам комиссии за 3 календарных днеи с даты проведения заседания, в некоторых случаях при необходимости в оперативных решении комиссии срок может быть продлен по решению председателя комиссии ( в его отсутствие заместителем председателя комиссии ) 37 итоги голосования членов комиссии оформляется протоколом в течение 3 календарных днеи с даты проведения заседания протокол оформляется председателем комиссии ( в его отсутствие заместителем председателя комиссии ) и секретарем комиссии при ее комиссиеи перед заочным голосованием протокол заседания оформляется в течение 3 календарных днеи с даты, установленнои для представления протокола бюллетенеи к протоколу прилагаются представленные членам комиссии бюллетени для голосования за члена комиссии, не представленные членам комиссии в установленныи для этого сроки либо заполненные ненадлежащим образом, при объявлении результатов голосования председатель и члены комиссии голосовавшие против принятого решения, а также воздержавшиеся должны в письменнои форме изложить свое особое мнение протокол приобщается к протоколу заседания комиссии протокол заседания комиссии должен содержать сведения о членах комиссии проголосовавших по вопросам повестки заседания, результатах голосования а также о принятых на заседании решениях, решение комиссии ( выписки из протокола заседания комиссии ) передается для исполнения в соответствующее структурное подразделение или акционерное общество, в течение 3 рабочих днеи с даты оформления копии протокола комиссия 4 права исполнять обязанности председателя, секретаря и членов комиссии. председателем комиссии является руководитель структурного подразделения, однои из основных задач которого является организация управления непрофильными активами [ORGANIZATION] и управление имуществом компании. ( возглавляет комиссию по общим экономическим и финансовым вопросам ) в отсутствие председателя комиссии обязанности председателя исполняет его заместитель председатель комиссии : организует работу комиссии и решение поставленных перед неи задач ; регулярно проводит заседания комиссии и председательствует на них ; обеспечивает широкое обсуждение рассматриваемых вопросов ; при необходимости дает поручения членам комиссии.'

In [ ]:
len(text.split())

In [157]:
augment_with_masked_lm(text)
# augment_with_masked_lm('[ORGANIZATION] ПРИКАЗ [DATE_TIME] No [DOCUMENT_NUMBER] Об утверждении Положений по непрофильным активам и порядке отчуждения непрофильных активов В целях приведения локальных нормативных актов Группы [ORGANIZATION] по вопросам управления непрофильными активами в соответствии с нормативными актами [ORGANIZATION] ПРИКАЗЫВАЮ: 1.')
# augment_with_masked_lm('[ORGANIZATION] ПРИКАЗ [DATE_TIME] No [DOCUMENT_NUMBER]. Об утверждении Положений по непрофильным')

0/249
1/262
2/262
3/262
4/262
5/262
6/262
7/262
8/262
9/262
10/262
11/262
12/262
13/262
14/262
15/262
16/262
17/262
18/262
19/262
20/262
21/262
22/262
23/262
24/262
25/262
26/262
27/262
28/262
29/262
30/262
31/262
32/262
33/262
34/262
35/261
36/261
37/261
38/261
39/261
40/261
41/261
42/261
43/261
44/261
45/261
46/261
47/261
48/261
49/261
50/260
51/260
52/260
53/260
54/260
55/260
56/259
57/259
58/259
59/259
60/259
61/259
62/259
63/259
64/259
65/258
66/258
67/258
68/257
69/257
70/257
71/256
72/256
73/255
74/255
75/255
76/255
77/255
78/255
79/255
80/254
81/254
82/254
83/254
84/254
85/254
86/253
87/253
88/253
89/253
90/253
91/252
92/252
93/252
94/252
95/252
96/252
97/252
98/252
99/251
100/251
101/251
102/251
103/250
104/250
105/250
106/250
107/249
108/249
109/249
110/249
111/249
112/249
113/249
114/249
115/249
116/249
117/249
118/249
119/249
120/249
121/249
122/248
123/248
124/248
125/248
126/248
127/248
128/248
129/248
130/248
131/248
132/248
133/248
134/248
135/248
136/248
137/248
138/24

'отменить от [DATE_TIME] a [DOCUMENT_NUMBER] об утверждении комиссии по непрофильным активам и порядке отчуждения таких активов в целях принятия новых правовых актов приказ [ORGANIZATION] по вопросам распоряжения такими активами и соответствия их деиствующим требованиям. приказываю : 1 отменить положение о комиссии по таким активам [ORGANIZATION] и организации групп компании, в новои редакции ( приложение 2. 2 отменить положение об условиях продажи таких активов ), организации групп компании, в новои редакцию, приложение 3. 3 отменить утверждение комиссии пo непрофильным активов, в ее отсутствие ( приложение 3. 4 признать утратившими силу ), от [DATE_TIME] a [DOCUMENT_NUMBER] и от нее no. 5 отменить деиствие деиствиев на организации группы компании, в которои компания имела возможность прямо или косвенно получить до 100 голосов ( приложение 3. 6 ) деиствие приказа возложить обязанности заместителя председателя правления – члена правления [PERSON] приложение 2 отменить приказу отменить 

In [25]:
a = '[ORGANIZATION] ПРИКАЗ [DATE_TIME] No [DOCUMENT_NUMBER] Об .утверждении Положений по непрофильным активам и порядке. отчуждения непрофильных. активов В целях приведения локальных нормативных актов .Группы [ORGANIZATION] по вопросам управления непрофильными. активами в соответствии .с нормативными актами [ORGANIZATION] ПРИКАЗЫВАЮ: 1. Утвердить Положение о Комиссии по. непрофильным активам [ORGANIZATION] и организаций. Группы компаний [ORGANIZATION]. в новой редакции (Приложение 1). 2. Утвердить Положение'
i_start = 0
splitted_text = a.split('.')
chunked_text = []
for i in range(len(splitted_text)):
    if sum(map(lambda x: x.count(' '), splitted_text[i_start:i+1])) > 10:
        chunked_text.append(''.join(splitted_text[i_start:i]))
        i_start = i
        
chunked_text


['[ORGANIZATION] ПРИКАЗ [DATE_TIME] No [DOCUMENT_NUMBER] Об ',
 'утверждении Положений по непрофильным активам и порядке отчуждения непрофильных',
 ' активов В целях приведения локальных нормативных актов ',
 'Группы [ORGANIZATION] по вопросам управления непрофильными активами в соответствии ',
 'с нормативными актами [ORGANIZATION] ПРИКАЗЫВАЮ: 1 Утвердить Положение о Комиссии по',
 ' непрофильным активам [ORGANIZATION] и организаций Группы компаний [ORGANIZATION]']

In [ ]:
'[ORGANIZATION] ПРИКАЗ [DATE_TIME] No [DOCUMENT_NUMBER] Об утверждении Положений по непрофильным активам и порядке отчуждения непрофильных активов В целях приведения локальных нормативных актов Группы [ORGANIZATION] по вопросам управления непрофильными активами в соответствии с нормативными актами [ORGANIZATION] ПРИКАЗЫВАЮ: 1. Утвердить Положение о Комиссии по непрофильным активам [ORGANIZATION] и организаций Группы компаний [ORGANIZATION] в новой редакции (Приложение 1). 2. Утвердить Положение о порядке отчуждения непрофильных активов [ORGANIZATION] и организаций Группы компаний [ORGANIZATION] в новой редакции (Приложение 2). 3. Утвердить состав Комиссии по непрофильным активам [ORGANIZATION] в новом составе (Приложение 3). 4. Признать утратившими силу приказы [ORGANIZATION] от [DATE_TIME] No [DOCUMENT_NUMBER] и от [DATE_TIME] No [DOCUMENT_NUMBER]. 5. Распространить действие настоящего Приказа на организации Группы компаний [ORGANIZATION], в которых Компания имеет право прямо или косвенно распоряжаться более 50 голосов (Приложение 4). 6. Контроль исполнения приказа возложить на заместителя Председателя Правления [PERSON] Председатель Правления [PERSON] Приложение 1 к приказу [ORGANIZATION] от [DATE_TIME] No [DOCUMENT_NUMBER] ПОЛОЖЕНИЕ О КОМИССИИ [ORGANIZATION] И ОРГАНИЗАЦИЙ ГРУППЫ КОМПАНИЙ [ORGANIZATION] ПО НЕПРОФИЛЬНЫМ АКТИВАМ 1. Общие положения Настоящее Положение определяет цели, задачи, организацию деятельности Комиссии [ORGANIZATION] по непрофильным активам (далее – Комиссия), права и обязанности Председателя, Секретаря и членов Комиссии, а также порядок представления отчетов о ходе ее работы. Требования настоящего Положения обязательны для применения при отчуждении непрофильных активов [ORGANIZATION] и организаций Группы компаний [ORGANIZATION]. Дочерние общества и организации [ORGANIZATION] обеспечивают утверждение локальных нормативных актов (положений), регламентирующих в соответствии с настоящим Положением порядок работы комиссий по непрофильным активам в дочерних обществах, а также в объектах вложений дочерних обществ, являющихся организациями Группы компаний [ORGANIZATION]. Представители интересов [ORGANIZATION] и дочерних обществ [ORGANIZATION] в органах управления организаций Группы компаний [ORGANIZATION] обеспечивают на основе настоящего Положения утверждение локальных нормативных актов (положений), регламентирующих порядок работы комиссий по непрофильным активам в таких организациях Группы компаний [ORGANIZATION]. 2. Основные цели и задачи Комиссии 2.1. Комиссия создается в целях обеспечения реализации Стратегии по управлению имуществом и иными активами [ORGANIZATION], Программы отчуждения непрофильных активов [ORGANIZATION], утверждаемой [ORGANIZATION], и Положения о комиссии [ORGANIZATION] по непрофильным активам. Комиссия в своей деятельности руководствуется законодательством Российской Федерации, решениями органов управления [ORGANIZATION] и [ORGANIZATION], иными внутренними документами, в том числе Положением о порядке отчуждения непрофильных активов [ORGANIZATION] и организаций Группы Газпром, Положением о комиссии [ORGANIZATION] по непрофильным активам, Положением о порядке отчуждения непрофильных активов [ORGANIZATION] и организаций Группы компаний [ORGANIZATION] и настоящим Положением. 2.2. Основной задачей Комиссии является принятие решений об отнесении активов, принадлежащих [ORGANIZATION] и организациям Группы компаний [ORGANIZATION], к категории непрофильных, а также определение порядка их реализации. В отношении активов, принадлежащих [ORGANIZATION] и организациям Группы компаний [ORGANIZATION], в случае, если их балансовая (остаточная) стоимость по данным бухгалтерского учета, или рыночная стоимость, или кадастровая стоимость (для земельных участков)1 составляет 100 (сто) и более миллионов рублей с учетом НДС решения принимает Комиссия [ORGANIZATION] по непрофильным активам. В отношении активов, принадлежащих [ORGANIZATION], а также организациям Группы компаний [ORGANIZATION] в случае, если их балансовая (остаточная) стоимость по данным бухгалтерского учета, или рыночная стоимость, или кадастровая стоимость (для земельных участков)1 составляет от 30 до 100 миллионов рублей с учетом НДС, решения принимаются Комиссией [ORGANIZATION]. В отношении активов, принадлежащих организациям Группы компаний [ORGANIZATION], в случае, если их балансовая (остаточная) стоимость по данным бухгалтерского учета, или рыночная стоимость, или кадастровая стоимость (для земельных участков)1 составляет менее 30 миллионов рублей с учетом НДС, решения принимаются Комиссией организации Группы компаний [ORGANIZATION]. 2.3. Для выполнения возложенной на нее задачи, Комиссия определяет критерии отнесения активов к категории непрофильных, а также принимает следующие решения: об отнесении активов к категории непрофильных; о признании непрофильных активов подлежащими/не подлежащими отчуждению; о целесообразности применения тех или иных способов отчуждения непрофильных активов; об определении условий отчуждения непрофильных активов2; о необходимости проведения предпродажной подготовки непрофильных активов, в том числе оценки их рыночной стоимости. 3. Организация деятельности Комиссии 3.1. Персональный состав Комиссии утверждается приказом [ORGANIZATION]. 3.2. Вопросы об отнесении активов к категории непрофильных вносятся на рассмотрение Комиссии по инициативе органов управления [ORGANIZATION], заместителей Председателя Правления и руководителей подразделений прямого подчинения Председателю Правления [ORGANIZATION], структурных подразделений [ORGANIZATION]. 3.3. Организация Группы компаний [ORGANIZATION] перед представлением материалов на рассмотрение Комиссии согласовывает обоснование целесообразности отчуждения актива, а также предполагаемого способа его отчуждения со структурными подразделениями [ORGANIZATION], ответственными за осуществление контроля за обеспечением эффективности долгосрочных финансовых вложений [ORGANIZATION] в соответствующие дочерние общества [ORGANIZATION], которым подконтрольна эта организация Группы компаний [ORGANIZATION] (далее - ответственное подразделение [ORGANIZATION]). Представление материалов на рассмотрение Комиссии осуществляется дочерними обществами [ORGANIZATION], в том числе в отношении подконтрольных организаций Группы компаний [ORGANIZATION], являющихся их объектами вложений, либо ответственным подразделением [ORGANIZATION]. 3.4 Комиссия по вопросам, входящим в ее компетенцию, имеет право: запрашивать у структурных подразделений [ORGANIZATION] и организаций Группы компаний [ORGANIZATION] необходимые для ее деятельности документы, материалы и информацию; устанавливать для структурных подразделений [ORGANIZATION] и организаций Группы компаний [ORGANIZATION] сроки и форму представления 1 В отношении объектов имущества, имеющих одни и те же идентификационные признаки (место нахождения, назначение и другие признаки), а также отчуждаемых в пользу одного приобретателя либо входящих в единый имущественный комплекс, рассчитывается суммарная балансовая (остаточная) или рыночная или кадастровая стоимость имущества (для земельных участков). 2 Во избежание дублирования норм и правил работы Комиссии и Инвестиционных комитетов Компании, вопросы о целесообразности отчуждения непрофильных активов могут быть приняты к рассмотрению Комиссией только после их рассмотрения на соответствующем Инвестиционном Комитете запрашиваемых документов, материалов и информации. 3.5. Заседания Комиссии проводятся по мере необходимости. Заседание Комиссии является правомочным при участии в нем не менее половины от общего числа ее членов. При очной форме проведения заседания Комиссии решения принимаются открытым голосованием простым большинством голосов членов Комиссии, присутствующих на заседании. При заочной форме проведения заседания Комиссии голосование осуществляется посредством бюллетеней, которые направляются ее членам в соответствии с настоящим Положением. Бюллетени для голосования должны содержать указание на дату представления заполненного бюллетеня. В случае равенства голосов решающим является голос Председателя Комиссии, а в его отсутствие заместителя Председателя Комиссии. 3.6. Повестка заседания Комиссии, письмо о созыве заседания с указанием формы, даты, времени и места его проведения, подписанные Председателем Комиссии (в его отсутствие заместителем Председателя Комиссии), информационные материалы и бюллетени для голосования (при проведении заседания в заочной форме) направляются членам Комиссии за 5 календарных дней до даты проведения заседания Комиссии. В исключительных случаях при необходимости принятия оперативных решений указанный срок может быть сокращен по решению Председателя Комиссии (в его отсутствие заместителя Председателя Комиссии). 3.7. Итоги голосования членов Комиссии оформляются протоколом в течение 3 календарных дней с даты проведения заседания. Протокол подписывается Председателем Комиссии (в его отсутствие заместителем Председателя Комиссии) и Секретарем Комиссии. При принятии Комиссией решений заочным голосованием протокол заседания оформляется в течение 3 календарных дней с даты, установленной для представления заполненных бюллетеней. К протоколу прилагаются подписанные членами Комиссии бюллетени для голосования. Бюллетень члена Комиссии, не представленный Секретарю Комиссии в установленный для голосования срок, либо заполненный ненадлежащим образом, при подведении итогов голосования не учитывается. Члены Комиссии, голосовавшие против принятого решения, а также воздержавшиеся вправе в письменной форме изложить свое особое мнение, которое приобщается к протоколу заседания Комиссии. Протокол заседания Комиссии должен содержать информацию о членах Комиссии, проголосовавших по вопросам повестки заседания, результатах голосования, а также о принятых на заседании решениях. 3.8. Решение Комиссии (выписки из протокола заседания Комиссии) направляются для исполнения в соответствующее структурное подразделение или дочернее общество [ORGANIZATION] в течение 3 рабочих дней с даты оформления соответствующего протокола Комиссии. 4. Права и обязанности Председателя, Секретаря и членов Комиссии 4.1. Председателем Комиссии назначается руководитель структурного подразделения [ORGANIZATION], одной из основных задач которого является обеспечение управления непрофильными активами [ORGANIZATION] и организаций Группы компаний [ORGANIZATION] (далее Департамент по правовым, корпоративным и имущественным вопросам). В отсутствие Председателя Комиссии обязанности Председателя исполняет его заместитель. Председатель Комиссии: организует работу Комиссии и выполнение возложенных на нее задач; созывает, проводит заседания Комиссии и председательствует на них; обеспечивает коллегиальное обсуждение рассматриваемых вопросов; при необходимости дает поручения членам Комиссии;'

In [160]:
labels = [('Блок технического директора', 199),
 ('Блок директора по мощностям', 194),
 ('Блок директора по строительству', 134),
 ('Управление по проектным работам', 109),
 ('Блок заместителя генерального директора по безопасности', 100),
 ('Генеральный директор', 82),
 ('Проект "Нефтяные краюшки"', 63),
 ('Блок деректора по газу', 58),
 ('Блок заместителя генерального директора по закупкам', 54),
 ('Блок заместителя генерального директора по организационным вопросам', 46),
 ('Проект сервиса скважин', 38),
 ('Блок директора по проектированию', 35),
 ('Проект "Новая нефть"', 30),
 ('Проект "Северная деревня"', 29),
 ('Блок операционного директора', 26),
 ('Блок директора по газовым проектам', 25),
 ('Блок заместителя генерального директора по защите', 21),
 ('Блок финансового директора', 21),
 ('Блок директора по портфелю', 19),
 ('Управление землеустроительных работ', 18),
 ('Проект "Трубопроводный транспорт Главного НГКМ"', 16),
 ('Блок директора по персоналу', 14),
 ('Проект «Обустройство объектов Новой нефти»', 14),
 ('Проект "Восточный"', 14),
 ('Блок заместителя генерального директора по имуществу', 14),
 ('Проект "Южный"', 10),
 ('Блок бизнес-директора', 6),
 ('Управление коммуникаций', 6),
 ('Проект «Обустройство Интересного лицензионного участка»', 6),
 ('Блок исполнительного директора по реализации проекта "Большое месторождение"',
  5),
 ('Проект "Обустройство площадных объектов НГКМ Поменбше"', 5),
 ('Подразделение по информационным технологиям', 2),
 ('Имущественные вопросы', 2),
 ('Проект «Трубопроводный транспорт Ещё одного НГКМ»', 2),
 ('Блок заместителя генерального директора по строительству', 2),
 ('Проект «Обустройство объектов Новейшей нейти»', 2)]

In [171]:
for i in range(len(df)):
    # print(df['label'][i])
    # print(list(filter(lambda x: x[0] == df['label'][i], labels))[0][1])
    if list(filter(lambda x: x[0] == df['label'][i], labels))[0][1] < 30:
        new_row = df.iloc[i].copy()
        new_row['idx'] = f'{new_row['idx']}_maskedLM_1'
        # print(new_row['idx'])
        try:
            new_row['text'] = augment_with_masked_lm(new_row['text'])
        except Exception as e:
            print(e)
            print(i)
            print(new_row['idx'])
            continue
        
        df = pd.concat([df, pd.DataFrame([new_row])], ignore_index=True)
# last_row = df.iloc[10].copy()
# print(last_row['idx'])
df.to_csv('augmented_marked_data')

The size of tensor a (820) must match the size of tensor b (512) at non-singleton dimension 1
52
1178_maskedLM_1
The size of tensor a (548) must match the size of tensor b (512) at non-singleton dimension 1
63
1198_maskedLM_1
The size of tensor a (525) must match the size of tensor b (512) at non-singleton dimension 1
64
1200_maskedLM_1
The size of tensor a (629) must match the size of tensor b (512) at non-singleton dimension 1
112
1266_maskedLM_1
The size of tensor a (561) must match the size of tensor b (512) at non-singleton dimension 1
114
1268_maskedLM_1
The size of tensor a (721) must match the size of tensor b (512) at non-singleton dimension 1
137
1302_maskedLM_1
The size of tensor a (513) must match the size of tensor b (512) at non-singleton dimension 1
177
1347_maskedLM_1
The size of tensor a (586) must match the size of tensor b (512) at non-singleton dimension 1
190
1364_maskedLM_1
The size of tensor a (606) must match the size of tensor b (512) at non-singleton dimension

The size of tensor a (820) must match the size of tensor b (512) at non-singleton dimension 1
52
1178_maskedLM_1
The size of tensor a (548) must match the size of tensor b (512) at non-singleton dimension 1
63
1198_maskedLM_1
The size of tensor a (525) must match the size of tensor b (512) at non-singleton dimension 1
64
1200_maskedLM_1
The size of tensor a (629) must match the size of tensor b (512) at non-singleton dimension 1
112
1266_maskedLM_1
The size of tensor a (561) must match the size of tensor b (512) at non-singleton dimension 1
114
1268_maskedLM_1
The size of tensor a (721) must match the size of tensor b (512) at non-singleton dimension 1
137
1302_maskedLM_1
The size of tensor a (513) must match the size of tensor b (512) at non-singleton dimension 1
177
1347_maskedLM_1
The size of tensor a (586) must match the size of tensor b (512) at non-singleton dimension 1
190
1364_maskedLM_1
The size of tensor a (606) must match the size of tensor b (512) at non-singleton dimension 1
210
14820_maskedLM_1
The size of tensor a (552) must match the size of tensor b (512) at non-singleton dimension 1
216
2010_maskedLM_1
The size of tensor a (541) must match the size of tensor b (512) at non-singleton dimension 1
223
4343_maskedLM_1
The size of tensor a (552) must match the size of tensor b (512) at non-singleton dimension 1
348
4518_maskedLM_1
The size of tensor a (692) must match the size of tensor b (512) at non-singleton dimension 1
602
5601_maskedLM_1
The size of tensor a (537) must match the size of tensor b (512) at non-singleton dimension 1
651
5696_maskedLM_1
The size of tensor a (541) must match the size of tensor b (512) at non-singleton dimension 1
688
5771_maskedLM_1
The size of tensor a (748) must match the size of tensor b (512) at non-singleton dimension 1
703
5811_maskedLM_1
The size of tensor a (547) must match the size of tensor b (512) at non-singleton dimension 1
721
5860_maskedLM_1
The size of tensor a (655) must match the size of tensor b (512) at non-singleton dimension 1
795
6190_maskedLM_1
The size of tensor a (532) must match the size of tensor b (512) at non-singleton dimension 1
823
6342_maskedLM_1
The size of tensor a (530) must match the size of tensor b (512) at non-singleton dimension 1
840
6394_maskedLM_1
The size of tensor a (680) must match the size of tensor b (512) at non-singleton dimension 1
841
6396_maskedLM_1
The size of tensor a (539) must match the size of tensor b (512) at non-singleton dimension 1
842
6409_maskedLM_1
The size of tensor a (527) must match the size of tensor b (512) at non-singleton dimension 1
845
6432_maskedLM_1
The size of tensor a (516) must match the size of tensor b (512) at non-singleton dimension 1
865
6533_maskedLM_1
The size of tensor a (542) must match the size of tensor b (512) at non-singleton dimension 1
866
6536_maskedLM_1
The size of tensor a (576) must match the size of tensor b (512) at non-singleton dimension 1
918
6745_maskedLM_1
The size of tensor a (724) must match the size of tensor b (512) at non-singleton dimension 1
1046
5905_maskedLM_1
The size of tensor a (517) must match the size of tensor b (512) at non-singleton dimension 1
1098
6078_maskedLM_1
The size of tensor a (551) must match the size of tensor b (512) at non-singleton dimension 1
1173
6361_maskedLM_1
The size of tensor a (542) must match the size of tensor b (512) at non-singleton dimension 1
1186
6411_maskedLM_1
The size of tensor a (516) must match the size of tensor b (512) at non-singleton dimension 1
1187
6412_maskedLM_1
The size of tensor a (514) must match the size of tensor b (512) at non-singleton dimension 1
1190
6415_maskedLM_1
The size of tensor a (519) must match the size of tensor b (512) at non-singleton dimension 1
1246
6610_maskedLM_1
The size of tensor a (638) must match the size of tensor b (512) at non-singleton dimension 1
1282
6825_maskedLM_1
The size of tensor a (596) must match the size of tensor b (512) at non-singleton dimension 1
1285
6833_maskedLM_1
The size of tensor a (576) must match the size of tensor b (512) at non-singleton dimension 1
1292
6887_maskedLM_1
The size of tensor a (520) must match the size of tensor b (512) at non-singleton dimension 1
1297
6906_maskedLM_1
The size of tensor a (549) must match the size of tensor b (512) at non-singleton dimension 1
1353
7081_maskedLM_1
The size of tensor a (593) must match the size of tensor b (512) at non-singleton dimension 1
1370
7134_maskedLM_1

In [1]:
df.to_csv('augmented_marked_data.csv')

NameError: name 'df' is not defined